typescript/infra/scripts/warp-routes/monitor-warp-routes-balances.ts

```diff
@@ -468,27 +563,89 @@ const getXERC20Limit = async (
   };
 };
 
+async function getTokenPriceByChain(
+  chain: ChainName,
+  tokenPriceGetter: CoinGeckoTokenPriceGetter,
+): Promise<number | undefined> {
+  try {
+    return tokenPriceGetter.getTokenPrice(chain);
+  } catch (e) {
+    logger.warn('Error getting token price', e);
+    return undefined;
+  }
+}
+
+async function getNativeTokenValue(
```

Comment:
I guess we have this so we don't need to specify the coingecko ID for native?

Reply:
For native token type we assume that the native has token will be the token used, is this a safe assumption?

Reply:
I think this is fine, extra context https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/4545#discussion_r1809145827